mailing_list/models.py

```diff
@@ -58,9 +67,44 @@ def get_or_create_or_delete_mailing_lists_for_canvas_course_id(self, canvas_cour
         canvas_sections = canvas_api_client.get_sections(canvas_course_id)
         mailing_lists_by_section_id = self._get_mailing_lists_by_section_id(canvas_course_id)
 
+        # get a list of the primary sections
+        primary_sections = [s['id'] for s in canvas_sections if s['sis_section_id'] is not None]
+
         overrides = kwargs.get('defaults', {})
         result = []
+
+        # if there is more than one primary section
+        # create a primary mailing list.
+        course_list = None
+        if len(primary_sections) > 1:
+            try:
+                course_list = MailingList.objects.get(canvas_course_id=canvas_course_id, section_id__isnull=True)
+            except MailingList.DoesNotExist:
+                create_kwargs = {
+                    'canvas_course_id': canvas_course_id,
+                    'section_id': None
+                }
+                create_kwargs.update(overrides)
+                course_list = MailingList(**create_kwargs)
+                course_list.save()
+
+        # if there is a course_list, append it to the result list so
+        # it can be used by the template.
+        if course_list:
+            result.append({
+                'id': course_list.id,
+                'canvas_course_id': course_list.canvas_course_id,
+                'section_id': course_list.section_id,
+                'name': canvas_api_client.get_course(canvas_course_id)['course_code'],
+                'address': course_list.address,
+                'access_level': course_list.access_level,
+                'members_count': len(course_list.members),
+                'is_primary_section': True,
+                'is_course_list': True,
+            })
```

Comment:
done. the block was removed, but a new block had to be added to allow the list to be exposed to the template.